File: Programming/B01_C/S1_Basics/S1_01_Basic_syntax.c
/*
	Basic syntax
*/

//T// Contents
//T// Basic syntax
//T// Variables, constants, literals

//T// compile any C file /path/to/file1.c in a terminal with
// gcc /path/to/file1.c -o /path/to/exec1
//T// run the compiled executable in a terminal with
// /path/to/exec1

//T// include directive
#include <stdio.h>

//T// define directive
#define LENGTH1 10

//T// main function, (), {}
int main ()
{
//T// int, identifiers, ",", ";"
	int a,b,c;

//T// =, 0x, 0octal, u, +
	a = 0x5p-2,b = 03u,c = a + b;

//T// printf, escapes \n and modifiers %d
	printf("Hello World!\nNum:%d\n",c);

//T// Variables, constants, literals

//T// const keyword
    const int LENGTH2 = 5;

//T// return keyword
	return 0;
}

File: Computer_science/B04_Java/S1_Basics/S1_01_Basic_syntax.java
/*
    Basic syntax
*/

//T// Contents
//T// Basic syntax
//T// Variables, constants, literals

# INSTALLATION

Install the JDK (Java Development Kit), which contains programs like `java` and `javac`.

``` {.bash in="in CLI"}
sudo apt install default-jdk
```

//T// compile any Java file /path/to/file1.java in a terminal with
// javac /path/to/file1.java
//T// the executable is called /path/to/file1.class
//T// run the compiled executable in a terminal with
// java /path/to/file1.class

//T// public class block
public class S1_01_Basic_syntax
{
//T// main() method where execution starts
    public static void main(String []args)
    {
//T// keywords and identifiers
        int a1 = 5;
//T// function call to print in terminal
        System.out.println("Hello World: " + a1);
    }
}